File: templatemethod/exercise/CustomVideoPreset.java
package templatemethod.exercise;

public class CustomVideoPreset extends VideoPreset {
    @Override
    protected void applyColorCorrection() {
        System.out.println("Apply my favorite color correction...");
    }

    @Override
    protected void renderVideo() {
        System.out.println("render video for FHD...");
    }
}

File: typescript/patterns/template/exercise/custom-video-preset.ts
import { VideoPreset } from "./video-preset";

export class CustomVideoPreset extends VideoPreset {
  protected override applyColorCorrection() {
    console.log("Apply my favorite color correction...");
  }

  protected override renderVideo() {
    console.log("render video for FHD...");
  }
}
